compiler/optimizer/OMRCFGSimplifier.cpp

```diff
@@ -169,6 +156,594 @@ bool TR_CFGSimplifier::simplifyBooleanStore()
+
+      if (storeNode->getOpCodeValue() == TR::astore && storeNode->isHeapificationStore())
+         insTree->getNode()->setHeapificationStore(true);
+
+      compareTree->insertBefore(insTree);
+      treeCursor = treeCursor->getNextTreeTop();
+      }
+
+   if (triangle1) {
+      _cfg->removeEdge(_block, _next1);
+      if (_block->getNextBlock() != _next2)
+         {
+         TR::Node *gotoNode = TR::Node::create(compareNode, TR::Goto, 0);
+         gotoNode->setBranchDestination(_next2->getEntry());
+         _block->append(TR::TreeTop::create(comp(), gotoNode));
+         }
+   } else {
+      _cfg->removeEdge(_block, _next2);
+      if (_block->getNextBlock() != _next2)
+         {
+         TR::Node *gotoNode = TR::Node::create(compareNode, TR::Goto, 0);
+         gotoNode->setBranchDestination(_next1->getEntry());
+         _block->append(TR::TreeTop::create(comp(), gotoNode));
+         }
+   }
+   TR::TransformUtil::removeTree(comp(), compareTree);
+   if (trace())
+      traceMsg(comp(), "End simplifyCondStoreSequence.\n");   
+   TR::DebugCounter::incStaticDebugCounter(comp(), TR::DebugCounter::debugCounterName(comp(), "cfgSimpMovSeq/%d/(%s)", count, comp()->signature()));
+   return true;
+   }
+
+bool OMR::CFGSimplifier::simplifySimpleStore(bool needToDuplicateTree)
+   {
+   static char *disableSimplifySimpleStore = feGetEnv("TR_disableSimplifySimpleStore");
+   if (disableSimplifySimpleStore != NULL)
+      return false;
+
+   if (trace())
+      traceMsg(comp(), "Start simplifySimpleStore block_%d\n", _block->getNumber());
+
+   TR::TreeTop *compareTree = _block->getLastRealTreeTop();
+   TR::Node *compareNode = compareTree->getNode();
+
+   bool triangle2 = _next2->getSuccessors().size() == 1 
+       && _next2->getExceptionSuccessors().size() == 0
+       && toBlock(_next2->getSuccessors().front()->getTo()) == _next1;
+   bool triangle1 = _next1->getSuccessors().size () == 1
+       && _next1->getExceptionSuccessors().size() == 0
+       && toBlock(_next1->getSuccessors().front()->getTo()) == _next2;
+   bool diamond = _next2->getSuccessors().size() == 1
+       && _next2->getExceptionSuccessors().size() == 0
+       && _next1->getSuccessors().size() == 1
+       && _next1->getExceptionSuccessors().size() == 0
+       && toBlock(_next1->getSuccessors().front()->getTo()) == toBlock(_next2->getSuccessors().front()->getTo());
+
+   if (trace())
+      {
+      traceMsg(comp(), "   block%d triangle1: %d triangle2: %d diamond: %d\n", _block->getNumber(), triangle1, triangle2, diamond);
+      }
+
+   static char *disableSimplifySimpleStoreTriangle = feGetEnv("TR_disableSimplifySimpleStoreTriangle");
+   if ((triangle1 || triangle2) && disableSimplifySimpleStoreTriangle != NULL)
+      return false;
+
+   static char *disableSimplifySimpleStoreDiamond = feGetEnv("TR_disableSimplifySimpleStoreDiamond");
```

Comment:
Is there a reason for using an environment variable instead of an option here?

Reply:
Commented previously.